src/auth/user-import-builder.ts

```diff
@@ -57,17 +70,8 @@ export interface UserImportRecord {
   phoneNumber?: string;
   photoURL?: string;
   disabled?: boolean;
-  metadata?: {
-    lastSignInTime?: string;
-    creationTime?: string;
-  };
-  providerData?: Array<{
-    uid: string;
-    displayName?: string;
-    email?: string;
-    photoURL?: string;
-    providerId: string;
-  }>;
+  metadata?: UserMetadataRequest;
+  providerData?: Array<UserProviderRequest>;
```

Comment:
These changes are mostly a side-effect of an earlier PR that got merged into master, I will update the `minimal-auth` branch to reflect the changes